app/controllers/api/works_controller.rb

```diff
@@ -97,22 +97,31 @@ def destroy
   end
 
   def index
-    works = Work.all
-    render json: works,
-      each_serializer: WorkSerializer
+    work_count = Work.count
+    works = Work.page(params[:page])
+    render json: {
+      works: ActiveModel::Serializer::CollectionSerializer.new(works, each_serializer: WorkSerializer),
+      work_count: work_count
+    }
   end
 
   def filtered_works
     parsed_query = CGI.parse(params[:search_params])
-    filtered_works = params[:search_params] == "" ?  Work.all : Work.where(parsed_query)
-    render json: filtered_works,
-      each_serializer: WorkSerializer
+    filtered_works = params[:search_params] == "" ?  Work.all.page(params[:page]) : Work.where(parsed_query).page(params[:page])
+    work_count = params[:search_params] == "" ?  Work.count : Work.where(parsed_query).count
+    render json: {
+      works: ActiveModel::Serializer::CollectionSerializer.new(filtered_works, each_serializer: WorkSerializer),
+      work_count: work_count
+    }
   end
 
   def filtered_artist_hidden
-    filtered_works = Work.joins(:artist).where("artists.hidden=false").select{|work| work.hidden==false}
-    render json: filtered_works,
-      each_serializer: WorkSerializer
+    filtered_works = Work.joins(:artist).where("artists.hidden" => false, "works.hidden" => false).page(params[:page])
```

Comment:
Might be a little cleaner to save `Work.joins(:artist).where("artists.hidden" => false, "works.hidden" => false)` in a local variable?